integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,910 @@
+
+      // Additional event on top og GA destination
+      Integration.productAddedToWishlist = Integration.productAddedToWishlistEnhanced;
+      Integration.productShared = Integration.productSharedEnhanced;
+      Integration.productsSearched = Integration.productsSearchedEnhanced;
+
+      // This mapping is for events which are not supported by segment and will
+      // be tracked by track method.
+      Integration.customEventsMapping = {
+        login: Integration.loggedInEnhanced,
+        sign_up: Integration.signedUpEnhanced,
+        exception: Integration.exceptionOccuredEnhanced,
+        timing_complete: Integration.timingCompletedEnhanced,
+        generate_lead: Integration.leadGeneratedEnhanced,
+        set_checkout_option: Integration.setCheckoutOptionEnhanced
+      };
+    }
+    /* eslint-enable */
+});
+
+/**
+ * Initialize.
+ *
+ * https://developers.google.com/gtagjs
+ *
+ * @api public
+ */
+
+GTAG.prototype.initialize = function() {
+  var config = [];
+  var that = this;
+  var gaOptions = this.options.gaOptions;
+  var gaWebMeasurementId = this.options.gaWebMeasurementId;
+  var gaWebAppMeasurementId = this.options.gaWebAppMeasurementId;
+  var awConversionId = this.options.awConversionId;
+  var dcFloodLightId = this.options.dcFloodLightId;
+  var accountId =
+    gaWebMeasurementId ||
+    gaWebAppMeasurementId ||
+    awConversionId ||
+    dcFloodLightId;
+  if (gaWebMeasurementId || gaWebAppMeasurementId) {
+    var gaSetting = {};
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      // To Set persistent values we need to use set instead of config
+      // https://developers.google.com/analytics/devguides/collection/gtagjs/setting-values
+
+      gaSetting.custom_map = merge(
+        gaOptions.dimensions,
+        gaOptions.metrics,
+        gaOptions.contentGroupings
+      );
+    }
+
+    // https://developers.google.com/analytics/devguides/collection/gtagjs/ip-anonymization
+    if (gaOptions.anonymizeIp) {
+      gaSetting.anonymize_ip = true;
+    }
+
+    // https://developers.google.com/analytics/devguides/collection/gtagjs/cookies-user-id
+    if (gaOptions.domain) {
+      gaSetting.cookie_domain = gaOptions.domain;
+    }
+
+    // https://developers.google.com/analytics/devguides/collection/analyticsjs/enhanced-link-attribution
+    if (gaOptions.enhancedLinkAttribution) {
+      gaSetting.link_attribution = true;
+    }
+
+    // https://support.google.com/optimize/answer/9183119?hl=en
+    if (gaOptions.optimize) {
+      gaSetting.optimize_id = gaOptions.optimize;
+    }
+
+    // https://support.google.com/analytics/thread/7741119?hl=en
+    if (gaOptions.sampleRate) {
+      gaSetting.sample_rate = gaOptions.sampleRate;
+    }
+
+    if (gaOptions.siteSpeedSampleRate) {
+      gaSetting.site_speed_sample_rate = gaOptions.siteSpeedSampleRate;
+    }
+
+    if (gaOptions.useGoogleAmpClientId) {
```

Comment:
Do you have a reference that `use_amp_client_id` is the right paramter?

Reply:
@briemcnally   I have referred setting from this reference: https://community.tealiumiq.com/t5/Client-Side-Tags/Google-Analytics-gtag-js-Tag-Setup-Guide/ta-p/30743